### ISSUE TITLE
failing private test case for level 5

### ISSUE THREAD

**Issue by PhilusTan** [nus-cs2030/2122-s2#28]
Im not sure why I cant pass the private test case in level 5 

Screenshots (if any)
![error](https://user-images.githubusercontent.com/99698387/153971114-faa9b676-1606-4717-bccd-eb2a298d862d.jpg)

**Comment by jace-han** [comment]
Hello! The ArrayIndexOutOfBounds exception is raised as you are trying to access the first element ("Index 0 out of bounds") when the list is empty ("for length 0"). I can't really help for the loaders (marked by red boxes) just from expected and your output though.

**Comment by ldkai** [comment]
Hi! I ran into the same error as you, you may want to check if your code still runs normally for an empty cruise list

**Comment by Seibell** [comment]
I ran into the same error too so I just added a check for an empty cruise list before running code for level5

**Comment by tanweiming00** [comment]
One of the private test cases includes running an empty cruise list to the method. So your code must take into account of that!!

**Comment by jace-han** [comment]
If one is running a for loop **iterating the cruise list** to "get" the cruises, would an empty cruise list throw an error? To my understanding, the for loop would just not be executed. Unless the iterator was set to <= list.size()

**Comment by shotnothing** [comment]
> If one is running a for loop **iterating through the cruise list** to "get" the cruises, would an empty cruise list throw an error? To my understanding, the for loop would just not be executed. Unless the iterator was set to <= list.size()

Yeah, the for loop is not executed, as it will immediately meet the condition < list.size(). 

It is specified the JLS:
https://docs.oracle.com/javase/specs/jls/se8/html/jls-14.html#jls-14.14.1.1

`If the (possibly unboxed) value of the Expression is false the first time it is evaluated, then the Statement is not executed.`

**Comment by undeser** [comment]
Hey, the error you're experiencing is most likely happening because it is looping through an empty iterable. For my code, I actually implemented an if-else that if the iterable is empty, it will add something into it first before iterating through the iterable. Alternatively, you could use a do-while loop. Hope this helps.

**Comment by weekiatism** [comment]
Codecrunch has a test case to check if the list is empty. Hence do check if the list is empty before looping through it.  Using a while loop condition will make your code more elegant as compared to for loops.
